viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            i = user_data["watchlist"].index(movie)
+            user_data["watchlist"].pop(i)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
+
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+
+    rating_list=[]
+    for movie in user_data["watched"]:
+        if movie["rating"]> 0:
+            rating_list.append(movie["rating"])
+    
+    rating_sum=sum(rating_list)
+    ave_rating=rating_sum/len(rating_list)
+    
+    return ave_rating
+    
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    frequent_genre_counter={}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in frequent_genre_counter:
+            frequent_genre_counter[genre]+=1
+        else:
+            frequent_genre_counter[genre]=1
+    max_genre = max(frequent_genre_counter.values())
+    for genre in frequent_genre_counter:
+        if frequent_genre_counter[genre]== max_genre:
+            return genre
+    
+
 # -----------------------------------------
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
+def get_unique_watched(user_data):
+    # user_data= watched
+    # input return user_data which is a dict
+    # output return list of dictionaries, (movies we've seen only
+
+    #=movies I saw
+    
+    # movies_i_saw=user_data["watched"]
+    # movies_they_saw=user_data["friends"]["watched"]
+    my_movies=[]
+    same_movies=[]
+    
+    for movie in user_data["watched"]:
+        for friend in user_data["friends"]:# friend is a dict with watched key referring to list of movies
+            if movie in friend["watched"]:
+                same_movies.append(movie)
+    for movies in user_data["watched"]:
+        if movies not in same_movies:
+            my_movies.append(movies)
+    return  my_movies
+
+def get_friends_unique_watched(user_data):
+    friend_movies=[]
+    # same_movies=[]
+    # for friend in user_data["friends"]:# friend is a dict with watched key referring to list of movies
+    #           for movie in user_data["watched"]:
+    #         if movie in friend["watched"]:
+    #             same_movies.append(movie)
+    #             print(f"{friend}")
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"] and movie not in friend_movies:
+                friend_movies.append(movie)
+                print(f"MOVIE{movie}")

```

Comment:
Good work on this function! I see this `print` statement was left in, which is usually, though not always, an unintentional debug statement. It's not a great idea to let reviewers find these, so good to get in the habit of reading your submitted PR to make sure it doesn't contain unwanted debug print statements. Totally fine once in a while, but good to be aware of.